### ISSUE TITLE
Archival narrows the numbering namespace, so both numbering guards pass on a collision

### ISSUE THREAD

**Issue by erguvendogukan** [rebelytics/one-skill-to-rule-them-all#54]
## Summary

Archival and observation numbering both operate on `log.md`, and the skill specifies them to run in an order where the first one silently narrows the namespace the second one checks. Both numbering guards pass and the appended entries still collide with archived ones.

## What happened

In one session, following the skill's own order — archive resolved entries at session start, then append new observations:

1. Archival moved 28 resolved entries (25 ACTIONED, 3 DECLINED) out of `log.md` into `archive/log-2026-08-21.md`. Those entries included observations 398, 399 and 400 — the three highest numbers in the log.
2. The active log's maximum was now 397.
3. The pre-write step computed the next free number as 398, checked `log.md` for a collision, and correctly found none.
4. Three entries were appended as 398, 399, 400.
5. The post-write verification counted occurrences of each number in `log.md` and got exactly 1 for each.

Every guard ran, every guard passed, and the log now contained three entries whose numbers were already in use by three different observations sitting in `archive/`. The collision was found only by manually comparing against the archive directory, which no step in the procedure calls for.

## Why the existing guards do not catch it

Both guards in **How to Log** read `log.md` only:

```bash
grep -o '### Observation [0-9]*' log.md | grep -o '[0-9]*' | sort -n | tail -1
```

**Archival on Write** moves entries to `archive/log-<date>.md`. After that move, `log.md` is no longer the full identifier namespace — but the guards still treat it as if it were. The narrowing is invisible at the point where either mechanism is specified, because archival reads as housekeeping and numbering reads as correctness, so nothing connects them.

This is not a race condition, so the existing pre-write/post-write pairing does not help. Both checks answer the same question, and the question itself has become too narrow.

## Suggested fix

Make the archive part of the numbering namespace in both guards:

```bash
# pre-write
MAX=$(cat log.md archive/*.md 2>/dev/null \
  | grep -o '### Observation [0-9]*' | grep -o '[0-9]*' | sort -n | tail -1)
N=$((MAX + 1))
HITS=$(cat log.md archive/*.md 2>/dev/null | grep -cE "^### Observation ${N}:")
[ "$HITS" -eq 0 ] || { echo "COLLISION on #${N}"; exit 1; }

# post-write
TOTAL=$(cat log.md archive/*.md 2>/dev/null | grep -cE "^### Observation ${N}:")
[ "$TOTAL" -eq 1 ] || echo "collision or missing entry for #${N}"
```

Note the POSIX `grep -o` form rather than `grep -oP`: the PCRE variant returns empty instead of erroring on non-PCRE grep builds, so a broken check reads exactly like a clean one — which the skill already warns about elsewhere.

It would also be worth stating explicitly in **Archival on Write** that observation numbers are never reused, and that the archive therefore remains part of the namespace after entries move.

## Environment

- Claude Code on Windows (git-bash), single session, no parallel writers
- Log at the time: 38 entries, 10 OPEN, highest number 400

**Comment by rebelytics** [comment]
Thanks for the order-dependency you found. v3.0.0, released today, dissolves the mechanism it lived in: the log is now one file per observation, and the next id is derived from three sources (active directory, archive, and a persistent `.id-floor` high-water mark), so archival can no longer narrow the namespace the numbering check reads.

Closing with v3.0.0: if anything similar survives the per-file format, please reopen or file a new issue.
